dockstore-client/src/main/java/io/dockstore/common/FileProvisioning.java

```diff
@@ -355,6 +369,29 @@ private void provisionInputFile(String parameterFilePath, String targetPath, Pat
         }
     }
 
+    /**
+     * Returns the first target path and its scheme in <code>targetPaths</code>
+     * whose scheme is handled by a plugin in <code>pluginList</code>.
+     *
+     * @param pluginList     a list of plugins
+     * @param targetPaths    a list of targetPaths
+     * @return the first targetPath and scheme with a supported scheme, or <code>Optional.empty()</code>
+     */
+    static Optional<ImmutablePair<String, String>> findSupportedTargetPath(List<ProvisionInterface> pluginList, List<String> targetPaths) {
+        for (String target : targetPaths) {
+            URI objectIdentifier = URI.create(target);    // throws IllegalArgumentException if it isn't a valid URI
```

Comment:
I think you should catch this and keep on looping if the exception is thrown. If for some reason there is one bad URI and one good one, you would never process the good one.